python/src/scipp/plot/plot.py

```diff
@@ -40,19 +45,20 @@ def _input_to_data_array(item, key=None):
     to_plot = {}
     if su.is_dataset(item):
         for name in sorted(item.keys()):
-            to_plot[name] = item[name]
+            _insert_if_not_0D(to_plot, name, item[name])
```

Comment:
I don't think you need that check at all, because all 0D variables are excluded below in `plot` with `if ndims > 0:`

Reply:
I think your right, but we would still keep the exception message and avoid the indexerror here https://github.com/scipp/scipp/blob/master/python/src/scipp/plot/plot.py#L183

Reply:
yes, we need to keep the check that `len(output) > 0:`